flink-core/src/main/java/org/apache/flink/types/parser/IntParser.java

```diff
@@ -38,19 +38,28 @@ public int parseField(byte[] bytes, int startPos, int limit, byte[] delimiter, I
 
 		final int delimLimit = limit-delimiter.length+1;
 
+		if (bytes.length == 0) {
```

Comment:
This check is not strictly necessary, IMO.
`bytes` is a larger byte array which is reused by the calling `GenericCsvInputFormat`.

To reduce the processing overhead of each field, I would omit the check (here and in the Long and Short parsers)